extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/DefaultTenantConfigResolver.java

```diff
@@ -81,7 +81,14 @@ private TenantConfigContext getTenantConfigFromTenantResolver(RoutingContext con
     }
 
     boolean isBlocking(RoutingContext context) {
-        return getTenantConfigFromConfigResolver(context, false) == null;
+        TenantConfigContext resolver = getTenantConfigFromConfigResolver(context, false);
+
+        if (resolver != null) {
+            // we always run blocking if refresh token is enabled even if the tenant was already resolved
```

Comment:
@pedroigor But what if the refresh is never even utilized ? Example the user will logout manually before the refresh is even needed ? I wonder if what Stuart said can be done somehow, switch to the blocking mode dynamically only for refresh.
@stuartwdouglas As Pedro said, the blocking vs event loop decision is done early, but we don't even know if refresh will be needed for a current request at that stage, is it possible to switch back to the blocking mode when the request is already running in the even loop ?

Reply:
I would say that it would be nice to have a way to run blocking operations from a mechanism just like we have for identity providers.

However, for this particular issue, I think we can leave as is and handle it separately. Probably in conjunction with the issue I mentioned when Vert.x OAuth is operating in the event loop.

Reply:
@pedroigor Absolutely, better handling of the blocking calls can be done later, but here it looks like the request becomes blocking even if no refresh will be done for the current call. But the refresh is healthy to have for every deployment anyway, but as soon as they add it it is a performance penalty :-) so I wonder if we can avoid it somehow. Lets see what @stuartwdouglas may suggest